tasks/WebSocketTask.cpp

```diff
@@ -0,0 +1,141 @@
+/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
+
+#include "WebSocketTask.hpp"
+
+#include <seasocks/PrintfLogger.h>
+#include <seasocks/WebSocket.h>
+#include <seasocks/Server.h>
+
+#include <json/json.h>
+
+#include <string.h>
+#include <iostream>
+
+using namespace controldev;
+using namespace seasocks;
+
+struct controldev::JoystickHandler : WebSocket::Handler {
+    WebSocket *connection = nullptr;
+    WebSocketTask *task = nullptr;
+    void onConnect(WebSocket *socket) override{
+        if (connection != nullptr){
+            connection->close();
+        }
+        Json::Value msg = true;
+        socket->send(msg.asString());
+        connection = socket;
+    }
+    void onData(WebSocket *socket, const char *data) override{
+        Json::Value msg;
+        msg = task->updateRawCommand(data);
+        connection->send(msg.asString());
+        if (msg.asBool() && task->controlling)
+            task->_raw_command.write(task->raw_cmd_obj);
+    }
+    void onDisconnect(WebSocket *socket) override{
+        connection = nullptr;
+    }
+};
+
+bool WebSocketTask::updateRawCommand(const char *data){
+    Json::Value jdata;
+    Json::CharReaderBuilder rbuilder;
+    std::unique_ptr<Json::CharReader> const reader(rbuilder.newCharReader());
+    std::string errs;
+
+    if (!reader->parse(data, data+std::strlen(data), &jdata, &errs)){
+        std::cout << errs << std::endl;
+        return false;
+    }
+
+    try{
+        raw_cmd_obj.axisValue.clear();
+        raw_cmd_obj.buttonValue.clear();
+
+        raw_cmd_obj.axisValue[0] = jdata["axes"][0].asDouble();
+        raw_cmd_obj.axisValue[1] = jdata["axes"][1].asDouble();
+        raw_cmd_obj.axisValue[2] = jdata["buttons"][6].asDouble() * 2 - 1;
```

Comment:
Isn't that (and the few other button-to-axis conversions) very specific to the Xbox controller ?

If it is, please make it generic through configuration parameters. Also, using floating-point math to generate a boolean is an avenue for errors (0.99999999999 will be zero). Use comparisons (`axisValue[2] = .... > 0`) should be enough since we assume it can only be zero or one.

If it is not, please add a comment that explains the reason for this mumbo-jumbo.